### PR TITLE
fix: signAndSubmitTransaction params

### PR DESCRIPTION
**Pull request by sanyu1225**
I have noticed that the interface[ is there](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-react/src/useWallet.tsx#L38), but it is not being given ```options```[ parameter.](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-react/src/WalletProvider.tsx#L78)

### CODE REVIEW

File: packages/wallet-adapter-react/src/useWallet.tsx [hunk 1]
```diff
@@ -24,6 +24,11 @@ export {
   isMobile,
 };
 
+export interface TransactionOptions {
```
Comment (0xmaayan) [inline packages/wallet-adapter-react/src/useWallet.tsx]: I think this interface can be placed in the `core` package [types file](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/types.ts) then we can export the type and use it in the react package.

File: packages/wallet-adapter-react/src/useWallet.tsx [hunk 2]
```diff
@@ -24,6 +24,11 @@ export {
...
+  max_gas_amount?: string;
+  gas_unit_price?: string;
```
Comment (0xmaayan) [inline packages/wallet-adapter-react/src/useWallet.tsx]:
```suggestion
  max_gas_amount?: Uint64,
  gas_unit_price?: Uint64,
```

File: packages/wallet-adapter-core/src/WalletCore.ts [hunk 3]
```diff
@@ -254,16 +254,22 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
...
   Sign and submit an entry (not bcs serialized) transaction type to chain.
   @param transaction a non-bcs serialized transaction
+  @param options max_gas_amount and gas_unit_limit
   @return response from the wallet's signAndSubmitTransaction function
   @throws WalletSignAndSubmitMessageError
   */
   async signAndSubmitTransaction(
-    transaction: Types.TransactionPayload
+    transaction: Types.TransactionPayload,
+    options?: {
```
Comment (0xmaayan) [inline packages/wallet-adapter-core/src/WalletCore.ts]: after creating the TransactionOptions type in the core package (see other comment), we can
```suggestion
    options?: TransactionOptions
```

**Review by 0xmaayan — CHANGES_REQUESTED** [review]
Thanks for catching it!

File: packages/wallet-adapter-react/src/useWallet.tsx (hunk 1 above)
Comment (sanyu1225) [reply packages/wallet-adapter-react/src/useWallet.tsx]: commit [8230ee7](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/8230ee71c9ad6a41e67bcd6a4d5c451125ab4d45)

File: packages/wallet-adapter-react/src/useWallet.tsx (hunk 2 above)
Comment (sanyu1225) [reply packages/wallet-adapter-react/src/useWallet.tsx]: commit [8230ee7](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/8230ee71c9ad6a41e67bcd6a4d5c451125ab4d45)

File: packages/wallet-adapter-core/src/WalletCore.ts (hunk 3 above)
Comment (sanyu1225) [reply packages/wallet-adapter-core/src/WalletCore.ts]: commit [8230ee7](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/8230ee71c9ad6a41e67bcd6a4d5c451125ab4d45)

File: packages/wallet-adapter-core/src/types.ts [hunk 4]
```diff
@@ -106,3 +108,8 @@ export interface SignMessageResponse {
   signature: string | string[]; // The signed full message
   bitmap?: Uint8Array; // a 4-byte (32 bits) bit-vector of length N
 }
+
+export interface TransactionOptions {
+  max_gas_amount?: Uint64;
```
Comment (0xmaayan) [inline packages/wallet-adapter-core/src/types.ts]: ohh yes sorry, let's simply use `bigint` then?

**Review by 0xmaayan — APPROVED** [review]
NICE!

File: packages/wallet-adapter-core/src/types.ts (hunk 4 above)
Comment (sanyu1225) [reply packages/wallet-adapter-core/src/types.ts]: haha no problem , done [f6263fb](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/f6263fbd008bb7cb58837237f91984564f4d5448)

File: packages/wallet-adapter-react/src/useWallet.tsx [hunk 5]
```diff
@@ -35,15 +36,15 @@ export interface WalletContextState {
...
-    options?: V
+    options?: TransactionOptions
   ): Promise<any>;
   signTransaction<T extends Types.TransactionPayload, V>(
     transaction: T,
-    options?: V
+    options?: TransactionOptions
   ): Promise<any>;
   signMessage(message: SignMessagePayload): Promise<SignMessageResponse | null>;
   signMessageAndVerify(message: SignMessagePayload): Promise<boolean>;
```
Comment (gregnazario) [inline packages/wallet-adapter-react/src/useWallet.tsx]: Can remove `V` now, I don't think it's used for anything, or V should be TransactionOptions

File: packages/wallet-adapter-react/src/WalletProvider.tsx [hunk 6]
```diff
@@ -74,29 +75,43 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
     }
   };
 
-  const signAndSubmitTransaction = async (
-    transaction: Types.TransactionPayload
+  const signAndSubmitTransaction = async <
+    T extends Types.TransactionPayload,
+    V extends TransactionOptions | undefined
```
Comment (gregnazario) [inline packages/wallet-adapter-react/src/WalletProvider.tsx]: I'm not a big typescript dev, so I don't know.  Is this standard practice to provide types that can be extended?

Comment (0xmaayan) [reply packages/wallet-adapter-react/src/WalletProvider.tsx]: it is better for flexibility and extensibility while making sure the base type is used and satisfied

File: packages/wallet-adapter-react/src/useWallet.tsx (hunk 5 above)
Comment (0xmaayan) [reply packages/wallet-adapter-react/src/useWallet.tsx]: yea.. we either use `TransactionOptions` or `V extends TransactionOptions`, I would say the later

File: packages/wallet-adapter-react/src/WalletProvider.tsx (hunk 6 above)
Comment (sanyu1225) [reply packages/wallet-adapter-react/src/WalletProvider.tsx]: Thank you for your valuable suggestions. commit  [1281dba](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/1281dba03db8c0e8140c1339573a27fd636c287a)

File: packages/wallet-adapter-react/src/useWallet.tsx (hunk 5 above)
Comment (sanyu1225) [reply packages/wallet-adapter-react/src/useWallet.tsx]: Thank you for your valuable suggestions. commit  [1281dba](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/1281dba03db8c0e8140c1339573a27fd636c287a)

File: packages/wallet-adapter-react/src/WalletProvider.tsx [hunk 7]
```diff
@@ -75,28 +76,33 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
   };
 
   const signAndSubmitTransaction = async (
-    transaction: Types.TransactionPayload
+    transaction: Types.TransactionPayload,
+    options?: TransactionOptions | undefined
```
Comment (0xmaayan) [inline packages/wallet-adapter-react/src/WalletProvider.tsx]: no need the `undefined` as this is an optional param

Comment (sanyu1225) [reply packages/wallet-adapter-react/src/WalletProvider.tsx]: Thanks, commit [6fd7c33](https://github.com/aptos-labs/aptos-wallet-adapter/pull/156/commits/6fd7c3390ead72f912a95e1080538f5bb79e2d8f)

**Comment by 0xmaayan** [comment]
@sanyu1225 thank you for fixing it!

**Comment by flamefalcon7** [comment]
awesome
we need that!

**Pull request merged**